### ISSUE TITLE
The closing brace of a Sass nested property written with a value is measured by no rule

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#570]
### What minimal example or steps are needed to reproduce the bug?

Sass spells a nested property in two ways, and `postcss-scss` parses them as two different node types. `font: { … }`, with no value, comes back as a `Rule` whose selector is `font:`. `font: 12px { … }`, with one, comes back as a `Declaration` carrying a block — the one declaration of any syntax this plugin reads that has children.

`indentation` measures the closing brace of a block behind `isRule(node) || isAtRule(node)`, so it never asks about the second spelling. The three `block-closing-brace-*-before` rules never see it either: they walk with `root.walkRules` and `root.walkAtRules`.

```scss
a {
	font: 12px {
		color: red;
			}
}
```

| input | warnings |
| --- | --- |
| `a {⏎\tfont: 12px {⏎\t\tcolor: red;⏎\t\t\t}⏎}` | none |
| `a {⏎\tfont: {⏎\t\tcolor: red;⏎\t\t\t}⏎}` | `4:4 Expected indentation of 1 tab` |

Both spellings say the same thing to Sass, and one of them is measured. Measured on `2b04870` under `@stylistic/scss/indentation`; the same two answers come back on `cdf1784`.

[#509](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/509) widened the gap without touching its cause. Where such a block is closed by an at-rule carrying neither a block nor a semicolon of its own, the run in front of the brace stands in that at-rule's `raws.between`, and `lastNodeHoldsTheBlockAfter` answers `true` for the nested property as it does for any other container — `getBlockAfter` hands the run back, and nobody asks:

| input | before #509 | after |
| --- | --- | --- |
| `a {⏎\tfont: 12px {⏎\t\t@include m⏎\t\t\t}⏎}` | none | none |
| `a {⏎\tfont: {⏎\t\t@include m⏎\t\t\t}⏎}` | none | `4:4 Expected indentation of 1 tab` |

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": ["@stylistic/stylelint-plugin"],
	"overrides": [
		{
			"files": ["**/*.scss"],
			"customSyntax": "postcss-scss",
			"rules": {
				"@stylistic/scss/indentation": "tab"
			}
		}
	]
}
```

### What did you expect to happen?

The closing brace of a nested property is measured whichever of the two spellings the author used.

### What actually happened?

Only the valueless spelling is, because only that one is parsed as a rule.

### Notes

Found while closing #509. The list of containers a brace rule admits is written out by node type in four rules, and this is the container every one of those lists forgets — the same omission [#387](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/387) had to close for the run behind a colon. Whether the answer is one more type in each list or one shared reading of «a node carrying a block» is the question to settle.
